src/main/java/com/yelp/nrtsearch/server/query/QueryNodeMapper.java

```diff
@@ -580,9 +576,12 @@ private static Query getPrefixQuery(PrefixQuery prefixQuery, IndexState state) {
       throw new IllegalArgumentException(
           "Field \"" + prefixQuery.getField() + "\" is not indexed with terms");
     }
-
     MultiTermQuery.RewriteMethod rewriteMethod =
         getRewriteMethod(prefixQuery.getRewrite(), prefixQuery.getRewriteTopTermsSize());
+    if (fieldDef instanceof PrefixQueryable prefixQueryable) {
+      return prefixQueryable.getPrefixQuery(prefixQuery, rewriteMethod);
+    }
+
```

Comment:
I would suggest rewriting this method to only use the `PrefixQueryable` interface. This means that `AtomFieldDef` would also need to implement it.